core/src/transport/memory.rs

```diff
@@ -227,7 +227,8 @@ impl Transport for MemoryTransport {
             return Err(TransportError::MultiaddrNotSupported(addr));
         };
 
-        DialFuture::new(port).ok_or(TransportError::Other(MemoryTransportError::Unreachable))
+        DialFuture::new(port)
+            .ok_or_else(|| TransportError::Other(MemoryTransportError::Unreachable))
```

Comment:
Did clippy actually suggest these? That is new to me. Normally it recommends the opposite, i.e. if there is no allocation, use `ok_or`.

Reply:
Also, this isn't flagged in other PRs. Can you please revert this (and others)?

Reply:
Yes, I had wrong toolchain when running clippy. Reverting these changes.